src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,95 @@
+        Object instance = createNewInstance(currentClass);
+        initializeDependencies(instance);
+        return instance;
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceOrComponentClass) {
+        if (interfaceOrComponentClass.isInterface()) {
+            return findComponentImplementation(interfaceOrComponentClass);
+        } else if (isComponent(interfaceOrComponentClass)) {
+            return interfaceOrComponentClass;
+        } else {
+            throw new RuntimeException("Class " + interfaceOrComponentClass.getName()
+                    + " is not annotated with @Component");
+        }
+    }
+
+    private Class<?> findComponentImplementation(Class<?> interfaceClass) {
+        Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+        interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+
+        Class<?> implementationClass = interfaceImplementations.get(interfaceClass);
+        if (implementationClass == null) {
+            throw new RuntimeException("No implementation found for interface "
+                    + interfaceClass.getName());
+        }
+
+        if (!isComponent(implementationClass)) {
+            throw new RuntimeException("Implementation class "
+                    + implementationClass.getName()
+                    + " is not annotated with @Component");
+        }
+
+        return implementationClass;
+    }
+
+    private boolean isComponent(Class<?> clazz) {
+        return clazz.isAnnotationPresent(Component.class);
+    }
+
+    private Object createNewInstance(Class<?> currentClass) {
+        if (instances.containsKey(currentClass)) {
+            return instances.get(currentClass);
+        }
+
+        try {
+            Constructor<?> constructor = currentClass.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(currentClass, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Can't create instance of " + currentClass.getName());
+        }
+    }
+
+    private void initializeDependencies(Object instance) {
+        Field[] declaredFields = instance.getClass().getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                field.setAccessible(true);
+                try {
+                    Object dependency = getInstance(field.getType());
+                    field.set(instance, dependency);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't inject dependency into field: "
+                            + field.getName(), e);
```

Comment:
When throwing an exception, add an informative message to it. Also, include the caught exception in the `RuntimeException` object you throw.